src/hooks/useFederatedSubgraphQuery.ts

```diff
@@ -0,0 +1,108 @@
+import { createClient, Operations, Queries } from "@olympusdao/treasury-subgraph-client";
+import { createHooks } from "@wundergraph/react-query";
+import { useEffect, useState } from "react";
+import { Environment } from "src/helpers/environment/Environment/Environment";
+
+const wgNodeUrl: string | undefined = Environment.getWundergraphNodeUrl();
+const client = createClient({
+  ...(wgNodeUrl ? { baseURL: wgNodeUrl } : {}), // Override the wundergraph client's API endpoint if the environment variable is present
+}); // Typesafe WunderGraph client
+
+type TokenRecordArray = Exclude<Queries["paginated/tokenRecords"]["response"]["data"], undefined>;
+export type TokenRecord = TokenRecordArray[0];
+
+type ProtocolMetricArray = Exclude<Queries["paginated/protocolMetrics"]["response"]["data"], undefined>;
+export type ProtocolMetric = ProtocolMetricArray[0];
+
+type TokenSupplyArray = Exclude<Queries["paginated/tokenSupplies"]["response"]["data"], undefined>;
+export type TokenSupply = TokenSupplyArray[0];
+
+export const { useQuery: useFederatedSubgraphQuery } = createHooks<Operations>(client);
+
+export const useTokenRecordsQuery = (startDate: string | null | undefined) => {
+  return useFederatedSubgraphQuery({
+    operationName: "paginated/tokenRecords",
```

Comment:
do you create each of these wundergraph operations manually?

Reply:
There's three of these operations... largely the same code between all three. Much simpler than the code previously in the frontend!

https://github.com/OlympusDAO/treasury-subgraph/blob/develop/apps/server/.wundergraph/operations/paginated/tokenRecords.ts

Reply:
Anything for me to do on this one?

Reply:
no, i was just curious how it worked.